astropy/wcs/wcs.py

```diff
@@ -255,14 +254,6 @@ def __init__(
         self.divergent = divergent
         self.slow_conv = slow_conv
 
-        if kwargs:
-            warnings.warn(
-                f"Function received unexpected arguments ({list(kwargs)}) these "
-                "are ignored but will raise an Exception in the "
-                "future.",
-                AstropyDeprecationWarning,
```

Comment:
Can we be sure it actually throws an exception now?

Reply:
Python will now raise a `TypeError`, we don't have to implement anything ourselves.